### PR TITLE
Added one feature selection technique.

### PR DESCRIPTION
**Pull request by parva-jain**
I have added one Feature Selection technique namely PearsonCorrelation along with test script and test data. 
Partial #30.

### CODE REVIEW

**Comment by parva-jain** [comment]
Please do let me know your reviews and if everything works fine, I'm looking forward to adding one more Class for the chi-squared feature selection technique in the filter_based_selection script.

**Comment by sidhu1012** [comment]
> Please do let me know your reviews and if everything works fine, I'm looking forward to adding one more Class for the chi-squared feature selection technique in the filter_based_selection script.

You can work on another PR while this one is in review.

File: datascientist/feature_selection/filter_based_selection.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import pandas as pd
+import numpy as np
+
+'''Note:- Only numerical or categorcial values encoded to numerical can be passed as features or target. Anything else data type will throw error if passed. '''
+
+class PearsonCorrelation:
+    
+    '''https://en.wikipedia.org/wiki/Pearson_correlation_coefficient'''
+    
+    '''Params:
```
Comment (sidhu1012) [inline datascientist/feature_selection/filter_based_selection.py:18]: Good work but follow numpy docstring guidelines.

File: datascientist/feature_selection/filter_based_selection.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        features:Features are individual independent variables that act as the input in your system.
+        Prediction models use features to make predictions.
+        
+        target : The target is whatever the output of the input variables.
+        It could be the individual classes that the input variables maybe mapped to in case of a classification problem or the output value range in a regression problem.     
+        '''
+    
+    def __init__(self,features,target):
+        
+        self.X = features
```
Comment (sidhu1012) [inline datascientist/feature_selection/filter_based_selection.py:28]: What's the expected type of features and target.

**Comment by sidhu1012** [comment]
@parva-jain Good work but can a new function be added to class which can return the best possible features? 
Please create a separate folder for csv.

File: datascientist/feature_selection/filter_based_selection.py (hunk 2 above)
Comment (parva-jain) [reply datascientist/feature_selection/filter_based_selection.py:28]: I have mentioned it on the test file.

**Comment by parva-jain** [comment]
> @parva-jain Good work but can a new function be added to class which can return the best possible features?
> Please create a separate folder for csv and try to follow pep guidelines as much as you can [#54 (comment)](https://github.com/Dacience/DataScientist/pull/54#issuecomment-737288533)

There is already a function that returns top n best features where n = 1 by default if num_feature is not passed by the user.

**Comment by parva-jain** [comment]
Can you exactly point out what things to add so that it is ready to be merged?

File: datascientist/feature_selection/filter_based_selection.py (hunk 2 above)
Comment (sidhu1012) [reply datascientist/feature_selection/filter_based_selection.py:28]: Mention it here in docstring

File: datascientist/feature_selection/filter_based_selection.py (hunk 1 above)
Comment (sidhu1012) [reply datascientist/feature_selection/filter_based_selection.py:18]: > There is already a function that returns top n best features where n = 1 by default if num_feature is not passed by the user.

Yeah got it!

**Comment by sidhu1012** [comment]
> Can you exactly point out what things to add so that it is ready to be merged?

Fix the docstring else LGTM!

**Comment by parva-jain** [comment]
> > Can you exactly point out what things to add so that it is ready to be merged?
> 
> Fix the docstring and make a separate folder for csv else LGTM!

Done!!

File: datascientist/feature_selection/filter_based_selection.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Dec  2 14:28:56 2020
+
+@author: parva
+"""
```
Comment (sidhu1012) [inline datascientist/feature_selection/filter_based_selection.py:6]: Remove these.

File: datascientist/feature_selection/test/Pearson_test.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Dec  2 14:34:22 2020
+
+@author: parva
+"""
```
Comment (sidhu1012) [inline datascientist/feature_selection/test/Pearson_test.py:6]: Remove these too!

File: datascientist/feature_selection/filter_based_selection.py (hunk 3 above)
Comment (parva-jain) [reply datascientist/feature_selection/filter_based_selection.py:6]: ok

**Comment by parva-jain** [comment]
> @parva-jain In docstring of every function before parameters please add one to two lines defining what the function would do. Apart from the review looks good to me.

done!!

File: datascientist/feature_selection/filter_based_selection.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            higher coefficient value is first element followed by the elements 
+            with lower values. 
+        
+        Returns
+        -------
+        cor_feature: List containing top feat_num features name.
+        '''
+        cor_list = []
+        feature_name = self.X.columns.tolist()
+        # cor_score = pd.DataFrame(columns = ['feature','cor_score'])
```
Comment (sidhu1012) [inline datascientist/feature_selection/filter_based_selection.py:98]: What's this? Remove it if unnecessary.

File: datascientist/feature_selection/filter_based_selection.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            the features with lower values. 
+        
+        Returns
+        -------
+        out_feature: DataFrame containing top feat_num features with all the 
+        data(rows).
+        '''
+        cor_list = []
+        feature_name = self.X.columns.tolist()
+        # cor_score = pd.DataFrame(columns = ['feature','cor_score'])
```
Comment (sidhu1012) [inline datascientist/feature_selection/filter_based_selection.py:132]: This too?

File: datascientist/feature_selection/filter_based_selection.py [hunk 7]
```diff
@@ -0,0 +1,140 @@
+
```
Comment (sidhu1012) [inline datascientist/feature_selection/filter_based_selection.py:1]: Remove this blank line.

**Comment by parva-jain** [comment]
did the said changes.

**Comment by sidhu1012** [comment]
LGTM!

**Comment by parva-jain** [comment]
Just like this, can I add the Chi-squared Correlation class in the feature_based_selection script and open another PR?

**Comment by sidhu1012** [comment]
> Just like this, can I add the Chi-squared Correlation class in the feature_based_selection script and open another PR?

Yes!

**Pull request merged**